buildSrc/subprojects/lifecycle/src/main/kotlin/gradlebuild/lifecycle.gradle.kts

```diff
@@ -127,21 +88,18 @@ fun needsToUseTestVersionsAll() = isRequestedTask(allVersionsCrossVersionTest)
     || isRequestedTask(allVersionsIntegMultiVersionTest)
     || isRequestedTask(soakTest)
 
-/**
- * Basic compile and check lifecycle tasks.
- */
-fun TaskContainer.registerEarlyFeedbackLifecycleTasks() {
+fun TaskContainer.registerEarlyFeedbackRootLifecycleTasks() {
     register(compileAllBuild) {
         description = "Initialize CI Pipeline by priming the cache before fanning out"
         group = ciGroup
-        dependsOn(":base-services:createBuildReceipt", "compileAll")
+        dependsOn(":base-services:createBuildReceipt")
     }
 
     register(sanityCheck) {
         description = "Run all basic checks (without tests) - to be run locally and on CI for early feedback"
         group = "verification"
         dependsOn(
-            "compileAll", ":docs:checkstyleApi", "codeQuality", ":internal-build-reports:allIncubationReportsZip",
+            ":docs:checkstyleApi", ":internal-build-reports:allIncubationReportsZip",
```

Comment:
We don't need to do it now, but eventually it would be better if we won't have to add a sanityCheck task to the root proejct at all. Instead the "sanityCheck" of the subprojects in question here should get extended in the places where the corresponding tasks --- like `allIncubationReportsZip ` -- are created.

Sam for compileAll/services:createBuildReceipt above.